chain/index/chain-indexing-overview-for-rpc-providers.MD

```diff
@@ -0,0 +1,218 @@
+
+#### Recommendations
+
+1. **Archival Nodes**: **Keep GC disabled** (`GCRetentionEpochs` = 0) to retain all indexed data.
+
+2. **Non-Archival Nodes**:  Set `GCRetentionEpochs` to match the amount of chain state your node retains 
+(*Example:* if your node is configured to retain 2 days of Filecoin chain state with the Splitstore, set `GCRetentionEpochs` to `retentionDays * epochsPerDay = 2 * 2880 = 5760`).
+**Warning:** Setting this value below the chain state retention period will degrade RPC performance and reliability because the Index will lack data for epochs still present in the chain state.  
+
+### Removed Options
+
+**Note: The following config options no longer exist in Lotus and have been removed in favor of the ChainIndexer config options explained above. They can be removed when upgrading to Lotus v1.31.0.**
+
+```toml
+[Fevm]
+EthTxHashMappingLifetimeDays = 0
+
+[Events]
+DisableHistoricFilterAPI = false
+DatabasePath = ""
+```
+
+## Upgrade Steps
+
+> **Note:** One can upgrade/downgrade between [pre-ChainIndexer](#previous-indexing-system) and [with-ChainIndexer](#chainindexer-indexing-system) Lotus versions without conflict because they persist state to different directories and don't rely on each other. No backup is necessary (but extra backups don't hurt).  
+
+Upgrading to the new `ChainIndexer` involves these steps:
+
+1. **Stop the Lotus Node**
+   - Stop your Lotus node before starting the upgrade and migration process.
+2. **Update Configuration**
+   - Modify your Lotus configuration to enable the `ChainIndexer` as described in the [`ChainIndexer Config` section above](#chainindexer-config).
+3. **Restart Lotus Node**
+   - Restart your Lotus node with the new configuration.
+   - The `ChainIndexer` will begin indexing **real-time chain state changes** immediately in the `${LOTUS_PATH}/chainindex` directory.
+   - **However, it will not automatically index any historical chain state (i.e., any previously existing chain state). To perform backfilling, please see the [`Backfill` section below](#backfill).**
+
+> **Note:** It's recommended to keep the [pre-ChainIndexer](#previous-indexing-system) state directory around until you've confirmed you don't need to [downgrade](#downgrade).  After sustained successful operations after the upgrade, the [pre-ChainIndexer](#previous-indexing-system) state directory can be removed to reclaim disk space.  
+
+## Backfill
+There is no migration from [pre-ChainIndexer indices](#previous-indexing-system) to the [ChainIndex](#chainindexer-indexing-system) because the [pre-ChainIndexer indices](#previous-indexing-system) are not fully reliable.  Instead one needs to index historical chain state (i.e., backfill) using one of the following tools.
+
+> **Note:** The ChainIndex will consume ~10GB of storage per month of tipsets (e.g., ~86400 epochs).  Ensure you have sufficient disk space before proceeding.
+
+### `ChainValidateIndex` JSON RPC API
```

Comment:
```suggestion
### `ChainValidateIndex` RPC API
```

or just "API" too, also see your uses of "JSON" below, they're not necessary and the JSON bit is transparent to many (most?) users

Reply:
This may have been me.  I went with this terminology because:
1. It matches website naming: https://docs.filecoin.io/reference/json-rpc and https://lotus.filecoin.io/reference/basics/api-access/#go-json-rpc-client
2. Distinguish from ETH RPC API

I agree that "JSON" as a qualifier isn't great.  In the fullness of time, I assume we'd want to call this "Filecoin Node API" (based on https://github.com/filecoin-project/FIPs/pull/1027 ), or maybe this is a Lotus-specific API so it should be called `Lotus Node API`?

Reply:
yeah, this won't be part of the "Common Node API", it'll never be exposed on the gateway for example

I'd be happy just with "API", but "RPC API" seems palatable too